src/posthog-core.ts

```diff
@@ -1370,11 +1370,28 @@ export class PostHog {
         this.surveys.renderSurvey(surveyId, selector)
     }
 
-    /** Checks the feature flags associated with this Survey to see if the survey can be rendered. */
+    /**
+     * Checks the feature flags associated with this Survey to see if the survey can be rendered.
+     *
+     * This method is deprecated because it's synchronous and won't return the correct result if the survey is not loaded.
+     * @deprecated Use `canRenderSurveyAsync` instead.
+     * @param surveyId The ID of the survey to check.
+     * @returns A SurveyRenderReason object indicating if the survey can be rendered.
+     */
     canRenderSurvey(surveyId: string): SurveyRenderReason | null {
         return this.surveys.canRenderSurvey(surveyId)
     }
 
+    /**
+     * Checks the feature flags associated with this Survey to see if the survey can be rendered.
+     * @param surveyId The ID of the survey to check.
+     * @param callback A callback function that will be called with the SurveyRenderReason object indicating if the survey can be rendered.
+     * @param forceReload If true, the survey will be reloaded from the server, Default: false
+     */
+    canRenderSurveyAsync(surveyId: string, callback: SurveyRenderReasonCallback, forceReload = false) {
```

Comment:
@pauldambra master of JS, any ideas if not just deprecating the older method and adding a new one? is the `Async` suffix a name that makes sense if its a callback method?

Reply:
funnily enough I think I'd expect `Sync` in the name if it took a callback and `Async` if it returned a promise

but as above we can use promises now
so maybe that helps figure out the path forwards?

so, you could add canRenderSurveysAsync which returns a promise instead